### ISSUE TITLE
llama2_v2 re-materializes every weight matrix on every token

### ISSUE THREAD

**Issue by hiraditya** [vx-lang/Vx#426]
`transformer()` is called once per token:

```vx
for pos in 0..steps {
  let logits : &mut f32 = transformer(token, pos, &config, &state, &weights, &all_weights);
```

and inside its `for l in 0..p.n_layers` loop it allocates a tensor per weight matrix and copies the weights into it element by element:

```vx
let mut t_wq = Tensor<f32>([dim, dim]);
let wq_off = w.wq + l * dim * dim;
for j in 0..dim * dim {
  t_wq[0][j] = all_weights[0][wq_off + j];
}
```

Five of these per layer — `rms_att_w`, `t_wq`, `t_wk`, `t_wv`, `t_wo`. The source offset depends on `l` alone. Nothing in any of it depends on `token` or `pos`, so the whole copy is invariant across the token loop and is redone from scratch every time.

## Scale

Per layer per token, the copied elements are roughly `2·dim² + 2·dim·kv_dim + dim`. For a 7B-class configuration (`dim = 4096`, `kv_dim = 4096`, `n_layers = 32`) that is about 2.1 billion element copies per token — near 8.6 GB read and 8.6 GB written, per token, to reconstruct data that did not change.

The checked-in test compiles rather than running a checkpoint (`RUN: vxc %s --action emit-mlir`), so nothing here is measured today. The shape of the program is what the issue is about: this is the reference the tensor path is written against, and it is written in a way no decoder would run.

## What it should be

The copies hoist out of the token loop entirely — the weight tensors are built once, before `for pos`, and `transformer` takes them. That is a change to the program rather than to the compiler.

Worth asking whether the compiler should have anything to say about it. A loop-invariant allocation-and-fill inside a hot loop is recognizable, and hoisting it is a standard motion; whether Vx wants to do that or to expect the programmer to is a design question, not an obvious yes.

Separately: the copy exists at all because `all_weights` is one flat `DynTensor<f32>` and each matrix is a window into it at an offset. A view type would make the copy unnecessary rather than merely hoisted — see Vx#400.

The out-of-extent indexing these same loops use is filed separately.
